packages/nouns-contracts/contracts/governance/NounsDAODelegation.sol

```diff
@@ -22,19 +22,31 @@ import { NounDelegationToken } from './NounDelegationToken.sol';
 
 library NounsDAODelegation {
     function isDelegate(address account, uint256[] memory tokenIds) external view returns (bool) {
+        NounDelegationToken dt = NounDelegationToken(ds().delegationToken);
+        require(
+            dt.lastTransferTimestamp(account) < block.timestamp,
+            'cannot use voting power updated in the current block'
+        );
+
         for (uint256 i = 0; i < tokenIds.length; i++) {
-            if (delegateOf(tokenIds[i]) != account) {
+            if (delegateOf(tokenIds[i], dt) != account) {
                 return false;
             }
         }
         return true;
     }
 
-    function delegateOf(uint256 tokenId) public view returns (address) {
-        address delegationOwner = NounDelegationToken(ds().delegationToken).ownerOfNoRevert(tokenId);
+    function delegateOf(uint256 tokenId, NounDelegationToken dt) public view returns (address) {
+        address delegationOwner = dt.ownerOfNoRevert(tokenId);
         if (delegationOwner != address(0)) return delegationOwner;
 
-        return ds().nouns.ownerOf(tokenId);
+        NounsTokenLike nouns = ds().nouns;
+        address nouner = nouns.ownerOf(tokenId);
```

Comment:
reminder that there's more gas optimization we want to do here.

pasting some POC code I did in discord:

```
    function isDelegate(address account, uint256[] memory tokenIds) external view returns (bool) {
        NounDelegationToken dt = NounDelegationToken(ds().delegationToken);

        bool hasDelegationTokens;
        bool hasNounTokens;
        NounsTokenLike nouns = ds().nouns;

        for (uint256 i = 0; i < tokenIds.length; i++) {
            address delegationOwner = dt.ownerOfNoRevert(tokenId);
            if (delegationOwner != address(0)) {
                if (delegationOwner != account) {
                    return false;
                }
                hasDelegationTokens = true;
            } else {
                if (nouns.ownerOf(tokenId) != account) {
                    return false;
                }
                hasNounTokens = true;
            }
        }

        if (hasDelegationTokens) {
            require(
                dt.lastTransferTimestamp(account) < block.timestamp,
                'cannot use voting power updated in the current block'
            );
        }

        if (hasNounTokens) {
            address nounsTokenDelegate = nouns.delegates(account);
            (uint32 fromBlock, ) = nouns.checkpoints(nounsTokenDelegate, nouns.numCheckpoints(nounsTokenDelegate) - 1);
            require(fromBlock < block.number, 'cannot use voting power updated in the current block');
        }

        return true;
    }
```